tests/resiliency_tests.rs

```diff
@@ -0,0 +1,314 @@
+            .all()
+            .unwrap()
+            .iter()
+            .for_each(|(_, proxy)| proxy.delete().unwrap());
+
+        TOXI_ADDR.get_or_init(|| toxi_addr);
+    });
+
+    let local_url = match std::env::var_os("MEMCACHED_ADDR") {
+        Some(v) => v.into_string().unwrap(),
+        None => "127.0.0.1:11211".to_string(), // use IPV4 so that it resolves to a single Server
+    };
+    let local_port = PROXY_PORT.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+
+    let toxi_addr = TOXI_ADDR.get().unwrap();
+    let toxic_local_addr = format!("{}:{}", toxi_addr.ip(), local_port);
+
+    let proxies = vec![ProxyPack::new(
+        format!("local-memcached-{}", local_port),
+        toxic_local_addr.clone(),
+        local_url.clone(),
+    )];
+
+    let toxiproxy_client = ToxiproxyClient::new(toxi_addr);
+    assert!(toxiproxy_client.is_running());
+
+    let proxies = toxiproxy_client.populate(proxies).unwrap();
+    let proxies = proxies
+        .into_iter()
+        .map(|proxy| ProxyDrop { proxy })
+        .collect();
+
+    (proxies, toxic_local_addr)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_succeeds_with_clean_client() {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let keys = vec!["clean-key1", "clean-key2", "clean-key3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values).collect();
+
+        let mut clean_client = rt.block_on(async {
+            async_memcached::Client::new("tcp://127.0.0.1:11211".to_string())
+                .await
+                .unwrap()
+        });
+
+        for key in &keys {
+            let _ = rt.block_on(async { clean_client.delete(key).await });
+            let result = rt.block_on(async { clean_client.get(key).await });
+            assert_eq!(result, Ok(None));
+        }
+
+        let result = rt.block_on(async { clean_client.set_multi(&kv, None, None).await });
+
+        assert!(result.is_ok());
+    }
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_errors_with_toxic_client_via_with_down() {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let (proxies, toxic_local_addr) = create_proxies_and_configs();
+
+        let toxic_local_url = "tcp://".to_string() + &toxic_local_addr;
+        let toxic_proxy = &proxies[0];
+
+        let keys = vec!["with-down-key1", "with-down-key2", "with-down-key3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values).collect();
+
+        let mut clean_client = rt.block_on(async {
```

Comment:
Why do you need both a clean client and a toxic client? The toxic client should work fine as long as you don't use `down` ?

Reply:
Using the toxic client for everything until you apply a toxic condition is fine, but in order to do the checks after using an actual toxic (i.e. a closure using `with_down()` or `with_limit_data()`) you need a clean client.  The toxic client is still in a compromised state at that point and trying to use it for further actions leads to out-of-bounds errors like this:

```
thread 'tests::test_set_multi_errors_on_downstream_with_toxic_client_via_limit_data' panicked at /Users/mark/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bytes-1.7.1/src/bytes_mut.rs:394:9:
split_to out of bounds: 8 <= 1
```